File: programming_paradigms/fucntional_programming/java_examples/FunctionalProgrammingEasy.java
package programming_paradigms.fucntional_programming.java_examples;

import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

public class FunctionalProgrammingEasy {
    public static void main(String[] args) {
        List<Integer> numbers = Arrays.asList(1, 2, 3, 4, 5);

        List<Integer> squaredNumbers = numbers.stream().map(n -> n * n).collect(Collectors.toList());

        System.out.println("Original numbers: " + numbers);
        System.out.println("Squared numbers: " + squaredNumbers);
    }
}

File: programming_paradigms/fucntional_programming/golang_examples/FunctionalProgrammingEasy.go
package golangexamples

import "fmt"

func main() {
	numbers := []int{1, 2, 3, 4, 5}

	squaredNumbers := make([]int, len(numbers))
	for i, n := range numbers {
		squaredNumbers[i] = n * n
	}

	fmt.Println("Original numbers:", numbers)
	fmt.Println("Squared numbers:", squaredNumbers)
}
